### ISSUE TITLE
Redshift AT TIME ZONE 

### ISSUE THREAD

**Issue by ConstantinoSchillebeeckx** [sqlfluff/sqlfluff#3084]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the following produces incorrect output
```sql
select
date_w_tz at TIME ZONE 'Etc/UTC' AS bar
from foo
```

### Expected Behaviour

Expected output
```sql
SELECT date_w_tz AT TIME ZONE 'Etc/UTC' AS bar,
FROM foo
```

### Observed Behaviour

Produces
```sql
select date_w_tzattimezone'Etc/UTC' as bar
from foo
```

### How to reproduce

<img width="1196" alt="image" src="https://user-images.githubusercontent.com/8518288/163189958-3c5e31f2-7c81-43be-ab23-c9aa79c1e7af.png">

Also tested locally with `0.12.0`

### Dialect

Redshift

### Version

* sqlfluff, version 0.12.0
* Python 3.9.7

### Configuration

```
target/
dbt_modules/
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
This was a tricky one, but was to do with the way RedShift redefined `ColumnReferenceSegment` meaning the parse tree wasn't quite correct. Submitted a fix in #3087
